Dispatch and You Typo, OnActivityCallbacks Documented, Info for Selling Free

Jump-Suit opened the pull request:
No Longer "Store SDK". Replaced respectively with the now in service: "Discord GameSDK".

Added Code Callback Examples to OnActivity! Used for allowing Ask to Join, Spectate, and Sending/Receiving of them respectfully now documented as they are used for the given Flows.

 Added in Information about Selling Free on Discord.

As requested in Discord GameSDK Server, I have written some documentation that *should* clear up anything regarding selling for free on discord.

Jump-Suit committed:
Fixed Some Typos I Introduced in last PR.

Changed how the variables are read slightly in but not noticeable on run.
OnMemberConnect
OnMemberDisconnect
OnMemberUpdate

Fixed a typo regarding the 3rd Variable in OnLobbyMessage for Byte[] data had the incorrect Number reference.

Jump-Suit committed:
Fixed Typo in OnSpeaking Params

OnSpeaking parameter callback list for lobbyId description was typoed, noticed that and fixed!

Commit:
Merge pull request #1 from discordapp/master

Update my Fork to Latest Master Release

Commit:
Discord Lobbies Search Methods Updated.

Have made changes to the names and code examples to better reflect the current state of how it is implemented with lobby searching.

Commit:
Merge pull request #2 from discordapp/master

(Update My Fork From Main Repo) Updates GameSDK Lobbies Docs Search. (#1103)

Commit:
No longer Store SDK

Changed in Dispatch, Dispatch and You: "Store SDK" -> "Discord GameSDK". For conformity.

Commit:
Added in Code for OnActivity Callbacks

Added in Code Example Changes for OnActivity Callbacks. Used for allowing Ask to Join, Spectate, and Sending/Receiving of them respectfully now documented.

Commit:
Quick Space Typo on my end.

Quick Space Typo on my end.

Commit:
Added in Information about Selling Free.

As requested in Discord GameSDK Server, I have written some documentation that *should* clear up anything regarding selling for free on discord.

Jump-Suit reviewed (commented):
Remove Line 75 in How_To_Get_Your_Game_On_Discord
Forgot to remove.

Commit:
Just so you know, FREE is an acceptable price!

Shadorc committed:
Document self_stream (#1108)

Thanks @Shadorc!

VisualEhrmanntraut committed:
Fixed problematic link (#1106)

Thanks @VisualDevelopment! 😺

msciotti commented on docs/game_sdk/Activities.md:
````diff
@@ -340,6 +341,41 @@ Fires when a user accepts a game chat invite or receives confirmation from Askin
+
+###### Example
+
+```cs
````
I think this code sample may have gotten messed up?

Jump-Suit replied:
Fixed!

msciotti commented on docs/game_and_server_management/How_to_Get_Your_Game_on_Discord.md:
```diff
@@ -69,7 +69,10 @@ Once you've completed all the items on your checklist, the `Get Approved` button
 - your server, to make sure you're abiding by our [Community Guidelines](https://discordapp.com/guidelines)
 
-If we need to see some changes, you'll receive an email directing you on your steps to success. Otherwise, you'll get **APPROVED!!!!!!!!!!** What's that mean? It means you can start charging for your game and making money! Don't spend it all in one place. Or do. Treat yourself.
+If we need to see some changes, you'll receive an email directing you on your steps to success. Otherwise, you'll get **APPROVED!!!!!!!!!!** What's that mean? It now allows for you to put your game up for Full Access! 'Coming Soon!' on your master branch will now be replaced with the beautiful 'Download' or 'Buy' Options! Allowing you to now have the freedom to distribute your game for free, or y ...
```
This will conflict with @Amorelandra changes - so just rebase from master and pick somethin' that sounds good!

Jump-Suit replied:
Resolved and created as a seperate Paragraph/ Information.

Commit:
Merge pull request #3 from discordapp/master

Update Fork with Latest Master Changes.
Rebased.

Jump-Suit committed:
Resolved FInal Issues.

Added in Information about Open Beta/Access Notes regarding price and other small information.

Fixed OnActivityJoin mising the callback itself.

Jump-Suit committed:
Merge branch 'master' of https://github.com/Jump-Suit/discord-api-docs

msciotti commented on docs/game_and_server_management/How_to_Get_Your_Game_on_Discord.md:
```diff
@@ -65,12 +65,14 @@ Your game looks wonderful. You've got a beautiful store page, and awesome game,
 
 If we need to see some changes, you'll receive an email directing you on your steps to success. Otherwise, you'll get **APPROVED!!!!!!!!!!** What's that mean? It means you can start charging for your game and making money! Don't spend it all in one place. Or do. Treat yourself. Or maybe you want your game to be **free!** Live your best life. We support you!
 
+ After that sweet succession in being **Approved**, celebrating with your community in this grand accomplishment. After you feel that *Open Beta* is over on your master branch you swap from Early Access to now Full Access! Show off your game to the world freely or with a slap-on-the-wrist add that price tag to let those know you have finally evolved! Unleash your next round of attacks at the fina ...
```
Needs a little grammar love:

```
After the sweet success of being **Approved**, take a moment to celebrate with your community. Now, get to work again! You can start charging for your game, or give it away for free. Live your best life. We support you.
```

Couple notes:

```
swap from Early Access to now Full Access
```

Approved games can still be early access, so they don't necessarily need to swap to Full Access.

```
with a slap-on-the-wrist add that price tag
```

"slap on the wrist" has a bit of colloquial context of meaning "getting in trouble", so not quite sure that applies here heh.

Jump-Suit replied:
Tweaked and added some information to clarify Early Access CAN still sell free. Changed wording for adding price tag and my input information paragraph!

msciotti commented on docs/game_sdk/Activities.md:
```diff
@@ -340,6 +341,44 @@ Fires when a user accepts a game chat invite or receives confirmation from Askin
+                Encoding.UTF8.GetBytes(String.Format("Hello, {0}!", user.Username)));
+        }
+		//Sends this off to a Activity callback named here as 'UpdateActivity' passing in the discord instance details and lobby details
+        UpdateActivity(discord, lobby);
```
Not sure what the `discord` object is here?

msciotti replied:
Oh gotcha, I see. You're passing the core into another function called `UpdateActivity()` and then the lobby object so that you can update the presence.

I think without that function actually here, it's a little confusing. I'd say either omit this line, or make a small example here of what that function does.

Jump-Suit replied:
Added in UpdateActivity Example to that.

Jump-Suit replied:
Didn't include joinActivitySecret though, since that gets set outside the scope of UpdateActivity mind you.

Commit:
Added in UpdateActivity Example

Added in UpdateActivity Example for OnActivityJoin to know where that information goes.

Commit:
Tweaked Getting Approved 

Updated My information with requested changes.

Commit:
Merge pull request #4 from discordapp/master

Merge Commits from Master to my PR

msciotti commented on docs/game_and_server_management/How_to_Get_Your_Game_on_Discord.md:
```diff
@@ -65,12 +65,14 @@ Your game looks wonderful. You've got a beautiful store page, and awesome game,
 
 If we need to see some changes, you'll receive an email directing you on your steps to success. Otherwise, you'll get **APPROVED!!!!!!!!!!** What's that mean? It means you can start charging for your game and making money! Don't spend it all in one place. Or do. Treat yourself. Or maybe you want your game to be **free!** Live your best life. We support you!
 
+ After that sweet success in being **Approved**, take a moment to celebrate with your community in this grand accomplishment. Now that you feel that *Open Beta* is over on your master branch you can swap from Early Access to now Full Access, not required to be **Approved** mind you! Show off your game to the world freely or with the addition of a price tag to accompany your masterpiece to let tho ...
```
I think everything is good with the PR other than this bit; it still reads a little confusing to me. Would it be alright if we just used my suggested edits and then we can merge?

msciotti replied:
> After the sweet success of being **Approved**, take a moment to celebrate with your community. Now, get to work again! You can start charging for your game, or give it away for free. Live your best life. We support you.

Jump-Suit replied:
Yeah sure! Sounds good! Lets me re-sync from Master, remove that bit and should be 100%!

Jump-Suit replied:
Removed! Let me know if this is ready!

Commit:
Merge pull request #5 from discordapp/master

Master Merge: Removed duplicate Data Model entry in Achievements.md (#1112)

Commit:
Removed my additional Info.

Removed my additional Info regarding Free and other small information.
